interfaces/kmod/backend.go

```diff
@@ -103,54 +99,27 @@ func (b *Backend) Remove(snapName string) error {
 	return err
 }
 
-// combineSnippets combines security snippets collected from all the interfaces
-// affecting a given snap into a de-duplicated list of kernel modules.
-func (b *Backend) combineSnippets(snapInfo *snap.Info, snippets map[string][][]byte) (content map[string]*osutil.FileState, modules []string, err error) {
-	content = make(map[string]*osutil.FileState)
-
-	for _, appInfo := range snapInfo.Apps {
-		for _, snippet := range snippets[appInfo.SecurityTag()] {
-			// split snippet by newline to get the list of modules
-			for _, line := range bytes.Split(snippet, []byte{'\n'}) {
-				l := bytes.TrimSpace(line)
-				// ignore empty lines and comments
-				if len(l) > 0 && l[0] != '#' {
-					modules = append(modules, string(l))
-				}
-			}
-		}
+func deriveContent(spec *Specification, snapInfo *snap.Info) (map[string]*osutil.FileState, []string) {
+	content := make(map[string]*osutil.FileState)
```

Comment:
This can be all short-circuited if there are no modules in the spec. You don't have to create the map or do anything else when that is the case. Most interfaces will not use this backend so it is worth doing.